docs/resource/README.md

```diff
@@ -102,7 +102,105 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
+  * `k8s.namespace.name`/`k8s.pod.name`/`k8s.container.name`, resulting in the input
+  `${telemetry.sdk.name}.${telemetry.sdk.language}.${k8s.namespace.name}.${service.name}.${k8s.pod.name}.${k8s.container.name}`
+  * `host.id`, resulting in the input `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}.${host.id}`
+- When the SDK is running in an environment where a `/etc/machine-id`
+  (see [MACHINE-ID(5)](https://www.freedesktop.org/software/systemd/man/machine-id.html))
+  is available, the machine-id should be used in the input for generating a UUID v5:
+  `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}.${machine.id}`
+- When the SDK is running on a Windows environment and there's a reasonable way to read
+  registry keys for the SDK, the registry key
+  `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Cryptography\MachineGuid` can be used in a
+  similar way to Linux' machine-id above.
+- When no other source is available the SDK MUST generate a value using UUID v1 or v4.
+  This would also typically be the case for service instances running on Kubernetes,
+  given that the pod's name and namespace are not unique enough to determine a service
+  instance's identity and the container name cannot easily be inferred: on pods with
+  multiple containers, the `service.instance.id` would yield the same results for all
+  containers, which is not desirable. And given that the services are ephemeral on
+  Kubernetes, the `service.instance.id` would change on each restart, being therefore
+  no different than a completely new UUID per process.
+
+Examples in Go, using the package "github.com/google/uuid" to generate the UUIDs:
+
+```go
+package main
+
+import (
+  "fmt"
+
+  "github.com/google/uuid"
+)
+
+func main() {
+  // fixed namespace for the purposes of this OTEP
+  ns := uuid.MustParse("4d63009a-8d0f-11ee-aad7-4c796ed8e320")
+
+  // prefix when no namespace is available
+  prefix := "${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}"
+
+  // prefix when a namespace is available
+  prefix = "${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}"
+
+  // actual prefix for the OTel Go SDK, for a service named "customers"
+  prefix = "opentelemetry.go.customers"
+
+  // the generated service.instance.id where a host.id is used
+  host := "graviola"
+  id := uuid.NewSHA1(ns, []byte(fmt.Sprintf("%s.%s", prefix, host))) // 17ffc8fd-6ed7-5069-a5fb-2fed78f5455f
+  fmt.Printf("v5 host id: %v\n", id)
+
+  // the generated service.instance.id where a typical kubernetes values are available
+  namespace := "accounting"
+  name := "vendors" // typically, the deployment name
+  pod := "vendors-pqr-jh7d2"
+  container := "some-sidecar"
+  input := fmt.Sprintf("opentelemetry.go.%s.%s.%s.%s", namespace, name, pod, container)
```

Comment:
Is `opentelemetry.go` the workload identifier mentioned above? if so, this appears to be flipped from the language:

> When a UUID v5 is generated, the input MUST be prefixed with
`${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}`, followed by the
the workload identifier, which should tentatively stable. This means that the same service yields the

Reply:
- ${telemetry.sdk.name} is `opentelemetry`
- ${telemetry.sdk.language} is `go`